lib/reporters/writers/console.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+/**
+ * Save timer references (mochajs/mocha#237).
+ */
+var setTimeout = global.setTimeout;
+
+/**
+ * Provides methods used for output by console-oriented reporters.
+ *
+ * @description
+ * Not meant to be used directly.
+ *
+ * @package
+ * @mixin ConsoleWriter
+ */
+var ConsoleWriter = function() {
+  var buffer = '';
+
+  /**
+   * Writes string to reporter output.
+   *
+   * @package
+   * @param {string} str - String to write to the reporter output.
+   * @returns {this}
+   * @chainable
+   */
+  this.write = function(str) {
+    if (str) {
+      buffer += str; // Store until `console.log` invoked
+    }
+    return this;
+  };
+
+  /**
+   * Writes newline-terminated string to reporter output.
+   *
+   * @package
+   * @param {string} str - String to write to the reporter output.
+   * @returns {this}
+   * @chainable
+   */
+  this.writeln = function(str) {
+    this.write(str);
+    console.log(buffer);
+    buffer = ''; // Reset
+    return this;
+  };
+
+  /**
+   * Invokes provided completion function.
+   *
+   * @description
+   * Writes any buffered output.
+   *
+   * @package
+   * @param {Function} cb - Callback invoked when cleanup done
+   */
+  this._done = function(cb) {
+    if (buffer) {
+      var delay = 1000; // One second
+
+      this.writeln();
+      setTimeout(cb, delay);
```

Comment:
Why does only console writer need a timeout? Why 1s?

Reply:
As this method is invoked immediately prior to `Mocha.run` finishing up, the delay is there to (hopefully) allow completion of flushing any buffered (i.e., unwritten) output to the screen. The '1s' time is arbirtrary for the moment; it's rather long for CPU-related tasks, but as this is IO-related I'm uncertain just _what_ a good value would be.

Be happy to change the comment to "arbitrary", as I usually would have...